core/common/buffer.hpp

```diff
@@ -14,292 +14,264 @@
+      OUTCOME_TRY(bytes, unhex(hex));
+      return outcome::success(BufferT(std::move(bytes)));
+    }
 
     /**
      * @brief return content of bytearray as string
      * @note Does not ensure correct encoding
      * @return string
      */
-    std::string toString() const;
+    std::string toString() const {
+      return std::string{begin(), end()};
+    }
 
     /**
      * @brief return content of bytearray as a string copy data
      * @note Does not ensure correct encoding
      * @return string
      */
-    std::string_view asString() const;
+    std::string_view asString() const {
+      return std::string_view(reinterpret_cast<const char *>(data()),
+                              size());  // NOLINT
+    }
 
     /**
      * @brief stores content of a string to byte array
      */
-    static Buffer fromString(const std::string_view &src);
+    static BufferT fromString(const std::string_view &src) {
+      return {src.begin(), src.end()};
+    }
 
    private:
-    std::vector<uint8_t> data_;
-
-    template <typename T>
-    Buffer &putRange(const T &begin, const T &end);
-  };
-
-  static const Buffer kEmptyBuffer{};
-
-  using BufferMutRef = std::reference_wrapper<Buffer>;
-  using BufferConstRef = std::reference_wrapper<const Buffer>;
-
-  class BufferView : public gsl::span<const uint8_t>,
-                     public boost::equality_comparable<Buffer>,
-                     public boost::less_than_comparable<Buffer> {
-   public:
-    BufferView() = default;
-
-    BufferView(const Buffer &buf) noexcept
-        : gsl::span<const uint8_t>{buf.data(),
-                                   static_cast<index_type>(buf.size())} {}
-
-    BufferView(gsl::span<const uint8_t> span) noexcept
-        : gsl::span<const uint8_t>{span} {}
-
-    std::string toHex() const;
-
-    BufferView &operator=(gsl::span<const uint8_t> span) noexcept {
-      *this = BufferView{span};
+    template <
+        typename OutIt,
+        typename InIt,
+        bool isIter = std::is_same_v<OutIt, typename Base::iterator>,
+        bool isConstIter = std::is_same_v<OutIt, typename Base::const_iterator>,
+        typename = std::enable_if_t<isIter or isConstIter>,
+        typename = std::enable_if_t<std::is_base_of_v<
+            std::input_iterator_tag,
+            typename std::iterator_traits<InIt>::iterator_category>>>
+    BufferT &putRange(const InIt &begin, const InIt &end) {
+      static_assert(sizeof(*begin) == 1);
+      insert(this->end(), begin, end);
       return *this;
     }
+  };
 
-    BufferView &operator=(Buffer &&buf) = delete;
+  template <size_t MaxSize>
+  inline std::ostream &operator<<(std::ostream &os,
+                                  const BufferT<MaxSize> &buffer) {
+    return os << BufferView(buffer);
+  }
 
-    bool operator==(const common::Buffer &buf) const noexcept {
-      return std::equal(begin(), end(), buf.begin(), buf.end());
-    }
+  //  using Buffer = BufferT<std::numeric_limits<size_t>::max()>;
```

Comment:
Probably better to put stricter limit to max size of buffer. 100mb or even smaller

Reply:
1. Now with **MaxSize == std::numeric_limits<size_t>::max()** all checks of size are disabled in compile time. Any other values will activate them.
2. **MaxSize** valuates by **element count**, not by bytes.